File: fedorova-t.v-kt-41-22/DatabaseService.vb
Imports fedorova_t.v_kt_41_22.Database
Imports fedorova_t.v_kt_41_22.Models

Public Class DatabaseService
    Private ReadOnly _context As TeacherDbContext

    Public Sub New(context As TeacherDbContext)
        _context = context
    End Sub

    Public Async Function AddTeacherAsync() As Task
        Dim degree = New AcademicDegree()
        Dim position = New Position()
        Dim department = New Department()

        _context.AcademicDegrees.Add(degree)
        _context.Positions.Add(position)
        _context.Departments.Add(department)
        Await _context.SaveChangesAsync()

        Dim teacher = New Teacher With {
    .AcademicDegreeId = degree.Id,
    .PositionId = position.Id,
    .DepartmentId = department.Id
}

        _context.Teachers.Add(teacher)
        Await _context.SaveChangesAsync()
    End Function
End Class

File: fedorova-t.v-kt-41-22/DatabaseService.cs
using fedorova_t.v_kt_41_22.Database;
using fedorova_t.v_kt_41_22.Models;
using Microsoft.EntityFrameworkCore;


public class DatabaseService
{
    private readonly TeacherDbContext _context;

    public DatabaseService(TeacherDbContext context)
    {
        _context = context;
    }

    public async Task AddTeacherAsync()
    {
        var degree = new AcademicDegree();
        var position = new Position();
        var department = new Department();

        _context.AcademicDegrees.Add(degree);
        _context.Positions.Add(position);
        _context.Departments.Add(department);
        await _context.SaveChangesAsync();

        var teacher = new Teacher
        {
            DegreeId = degree.Id,
            PositionId = position.Id,
            DepartmentId = department.Id
        };

        _context.Teachers.Add(teacher);
        await _context.SaveChangesAsync();
    }
}
